src/shared/lib/StoreUtils.ts

```diff
@@ -479,21 +479,24 @@ export function mergeMutationsIncludingUncalled(mutationData:MobxPromise<Mutatio
 {
     const idToMutations: {[key: string]: Mutation[]} = {};
 
-    updateIdToMutationsMap(idToMutations, mutationData, generateMutationId);
-    updateIdToMutationsMap(idToMutations, uncalledMutationData, generateMutationId);
+    updateIdToMutationsMap(idToMutations, mutationData, generateMutationId, false);
+    updateIdToMutationsMap(idToMutations, uncalledMutationData, generateMutationId, true);
 
     return Object.keys(idToMutations).map((id:string) => idToMutations[id]);
 }
 
 function updateIdToMutationsMap(idToMutations: {[key: string]: Mutation[]},
                                 mutationData:MobxPromise<Mutation[]>,
-                                generateMutationId:MutationIdGenerator = generateMutationIdByGeneAndProteinChangeAndEvent)
+                                generateMutationId:MutationIdGenerator = generateMutationIdByGeneAndProteinChangeAndEvent,
+                                onlyUpdateIfExists: boolean)
 {
     if (mutationData.result) {
         for (const mutation of mutationData.result) {
             const mutationId = generateMutationId(mutation);
-            idToMutations[mutationId] = idToMutations[mutationId] || [];
-            idToMutations[mutationId].push(mutation);
+            if (!onlyUpdateIfExists || (onlyUpdateIfExists && (mutationId in idToMutations))) {
```

Comment:
You don't need the check if `onlyUpdateIfExist` on the right side of OR, since you are already checking if not `onlyUpdateIfExists` on the left side of OR :) When `onlyUpdateIfExists` is false, left hand side of the condition will always be true ;) So let's simplify this to:
`!onlyUpdateIfExists || (mutationId in idToMutations)`